checked/debug.go

```diff
@@ -44,7 +44,11 @@ var (
 )
 
 var tracebackCallersPool = sync.Pool{New: func() interface{} {
-	return make([]uintptr, tracebackMaxDepth)
+	// Pools should generally only return pointer types, since a pointer
+	// can be put into the return interface value without an allocation.
+	// Consequently, we return a pointer to a slice instead of just a slice.
+	pc := make([]uintptr, tracebackMaxDepth)
+	return &pc
```

Comment:
I don't think this is the right thing to do based on the usage pattern, see below.